aQute.libg/src/aQute/libg/classloaders/ModifiableURLClassLoader.java

```diff
@@ -0,0 +1,64 @@
+package aQute.libg.classloaders;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Enumeration;
+
+public class ModifiableURLClassLoader extends URLClassLoader {
+
+	private final ClassLoader lastLoader;
```

Comment:
This is a big behavior change from before. The current code using URLClassLoader uses the argument class loader as the parent (parent-first). You have changed it here to last. Not sure that is wise? And it requires odd behavior when used below.

Reply:
the point is that the parent classloader MUST NOT load the launcher class. Therefore making it parent first would break the fix entirely.

Reply:
Then it would probably be better to configure the class loader which a set of class names which must not be delegated to the parent class loader RATHER than parent-last behavior which causes the other weird issues below.

Reply:
perhaps.. I can try that.